docs/library/machine.Pin.rst

```diff
@@ -3,16 +3,42 @@
 class Pin -- control I/O pins
 =============================
 
-A pin is the basic object to control I/O pins (also known as GPIO -
-general-purpose input/output). It has methods to set
-the mode of the pin (input, output, etc) and methods to get and set the
-digital logic level. For analog control of a pin, see the ADC class.
+A pin object is used to control I/O pins (also known as GPIO - general-purpose
+input/output).  Pin objects are commonly associated with a physical pin that can
+drive a voltage and read voltages.  The pin class has methods to set the mode of
+the pin (IN, OUT, etc) and methods to get and set the digital logic level.
+For analog control of a pin, see the ADC class.
 
-Usage Model:
+A specific pin is constructed by using its identifier, which is specific to a
+given port.  Identifiers can always be integers and on some ports strings can be
```

Comment:
While I usually agree with no-duplication recommendations, in this case, I don't. As Pin class is the core of hardware functionality, it makes sense to make sure a user has an extra chance to not miss an important information.